Allow config to create projects with no existing geoms (new geom only)

spwoodcock opened the pull request:
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation
- [ ] 🧑‍💻 Refactor
- [ ] ✅ Test
- [ ] 🤖 Build or CI
- [ ] ❓ Other (please specify)

## Related Issue

Fixes #2270

## Describe this PR

- Added extra option to create a project with no geometries.
- For projects when only new geom needs to be collected.
- It's a bit messy / not perfect - I did this in a rush - but project creation works.
- Needs to be tested thoroughly.

## Checklist before requesting a review

- 📖 Read the FMTM Contributing Guide: <https://github.com/hotosm/fmtm/blob/main/CONTRIBUTING.md>
- 📖 Read the HOT Code of Conduct: <https://docs.hotosm.org/code-of-conduct>
- 👷‍♀️ Create small PRs. In most cases, this will be possible.
- ✅ Provide tests for your changes.
- 📝 Use descriptive commit messages.
- 📗 Update any related documentation and include any relevant screenshots.

## [optional] What gif best describes this PR or how it makes you feel?

Sujanadh commented:
For no data extracts, we can allow them to use split by square or choose project AOI as a task. This kind of project doesn't need to create entity; we can create dataset during project creation and avoid entity creation. I see your inline comment syaing status need to be set to avoid downstream error, what does that mean? failure of status api?

spwoodcock commented:
> For no data extracts, we can allow them to use split by square or choose project AOI as a task. This kind of project doesn't need to create an entity; we can create a dataset during project creation and avoid entity creation. I see your inline comment saying status needs to be set to avoid downstream errors; what does that mean? failure of status API?

Yeah I think so - having no status for an entity causes an error when its serialised as a JSON.

And yep I agree split by square or selecting the project AOI are the best options for this type of project. Interestingly, split by sql task splitting also worked, as it downloaded the osm data extract and split on that 😄

If avoiding the creation of an existing entities works and causes no errors, then that is a good option! 👍

Sujanadh commented:
Oh yes, splitting works :rofl:. I guess so; not creating entities is possible, but the dataset needs to be there.

spwoodcock commented:
@Sujanadh did you manage to check if creating a project without any existing entities works?

Let me know if this PR will be modified 🙏

Sujanadh commented:
yes i tested project creation without entity creation and it was working. But the mapper frontend wasn't working it seemed like some issue related to the svelte package?

Today i am on leave so can't be involved directly.

Sujanadh commented on src/frontend/src/api/CreateProjectService.ts:
```diff
@@ -41,7 +42,9 @@ const CreateProjectService = (
       if (!hasAPISuccess) {
-        throw new Error(`Request failed with status ${projectCreateResp.status}`);
+        const msg = `Request failed with status ${projectCreateResp.status}`;
+        console.error(msg);
```
do we need consoles here?

spwoodcock replied:
Yeah I think it makes sense, as it's quite a major error if the project creation fails. It can help debugging what went wrong

Sujanadh commented:
@spwoodcock after you review, this PR is good to go
